lib/sequence_diagram/sequencer.rb

```diff
@@ -0,0 +1,72 @@
+require "ostruct"
+
+module SequenceDiagram
+  class Sequencer
+    def self.sequence
+      sequencer = new
+      sequencer.enable { yield }
+      sequencer.sequence
+    end
+
+    def enable
+      trace.enable { yield }
+    end
+
+    def sequence
+      format_steps(steps)
+    end
+
+  private
+
+    def trace
+      @trace ||= TracePoint.new(:call, :return) do |trace_point|
+        capture(trace_point)
+      end
+    end
+
+    def callers
+      @callers ||= ["caller"]
+    end
+
+    def steps
+      @steps ||= []
+    end
+
+    def capture(trace_point)
+      steps << extract_step(trace_point)
+    end
+
+    def extract_step(trace_point)
+      OpenStruct.new(
+        caller:   determine_caller(trace_point),
+        receiver: trace_point.defined_class,
+        message:  trace_point.method_id,
+        event:    trace_point.event
+      )
+    end
+
+    def format_steps(steps)
+      ["sequenceDiagram"] + steps.map { |step| format_step(step) }
+    end
+
+    def format_step(step)
```

Comment:
I could see a separate class being devoted to the formatting concerns, cleaning up the core logic a little.